FlightStreamDeck.Logics/Actions/CustomGaugeAction.cs

```diff
@@ -0,0 +1,194 @@
+﻿using FlightStreamDeck.Core;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using SharpDeck;
+using SharpDeck.Events.Received;
+using SharpDeck.Manifest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FlightStreamDeck.Logics.Actions
+{
+    public class CustomGaugeSettings
+    {
+        public string HeaderTop { get; set; }
+        public string HeaderBottom { get; set; }
+        public float MinValue { get; set; }
+        public float MaxValue { get; set; }
+        public string ToggleValue { get; set; }
+        public string DisplayValueTop { get; set; }
+        public string DisplayValueBottom { get; set; }
+        public bool DisplayHorizontalValue { get; set; }
+        public string ChartSplitValue { get; set; }
+        public int ChartThicknessValue { get; set; }
+        public int ChartChevronSizeValue { get; set; }
+
+        internal bool EmptyPayload { get =>
+                string.IsNullOrEmpty(HeaderTop) &&
+                string.IsNullOrEmpty(HeaderBottom) &&
+                string.IsNullOrEmpty(ToggleValue) &&
+                string.IsNullOrEmpty(DisplayValueTop) &&
+                string.IsNullOrEmpty(DisplayValueBottom) &&
+                string.IsNullOrEmpty(ChartSplitValue) &&
+                MinValue == 0 &&
+                MaxValue == 0 &&
+                ChartThicknessValue == 0 &&
+                ChartChevronSizeValue == 0 && !DisplayHorizontalValue;
+        }
+    }
+
+    [StreamDeckAction("tech.flighttracker.streamdeck.custom.gauge")]
+    public class CustomGaugeAction : StreamDeckAction<CustomGaugeSettings>
+    {
+        private readonly ILogger<CustomGaugeSettings> logger;
+        private readonly IFlightConnector flightConnector;
+        private readonly IImageLogic imageLogic;
+        private readonly EnumConverter enumConverter;
+
+        private TOGGLE_EVENT? toggleEvent = null;
+        private TOGGLE_VALUE? displayValueTop = null;
+        private TOGGLE_VALUE? displayValueBottom = null;
+
+        private string currentValueTop = "0";
+        private string currentValueBottom = "0";
+
+        public CustomGaugeSettings settings = new CustomGaugeSettings() { 
```

Comment:
Inconsitent { placement. Do we { on newlines or not @nguyenquyhy ?

Reply:
I would prefer { on new line (unless everything on the same line) as my VS will auto format like that anyway.

Reply:
Got it, Its visual flavor. But I'll conform. :)

Reply:
True, it's all visual flavor. Same for the parameter grouping. But having a consistent choice (and I don't mind too much which choice) makes the code more readable and tidy :)